tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,195 @@
+            #   with --repeat
+            self.rnd_seed = int(time.time() * 1_000_000_000) + hash(
+                self.redpanda.service_id)
+        else:
+            self.rnd_seed = rnd_seed_override
+        self.logger.info(f"Random seed: {self.rnd_seed}")
+        self.rnd = random.Random(self.rnd_seed)
+        self.rpk = RpkTool(self.redpanda)
+
+    class ConfigProp(Enum):
+        QUOTA_NODE_MAX_IN = "kafka_throughput_limit_node_in_bps"
+        QUOTA_NODE_MAX_EG = "kafka_throughput_limit_node_out_bps"
+        THROTTLE_DELAY_MAX_MS = "max_kafka_throttle_delay_ms"
+        BAL_WINDOW_MS = "kafka_quota_balancer_window_ms"
+        BAL_PERIOD_MS = "kafka_quota_balancer_node_period_ms"
+        QUOTA_SHARD_MIN_RATIO = "kafka_quota_balancer_min_shard_throughput_ratio"
+        QUOTA_SHARD_MIN_BPS = "kafka_quota_balancer_min_shard_throughput_bps"
+
+    def get_config_parameter_random_value(self, prop: ConfigProp):
+        if prop in [
+                self.ConfigProp.QUOTA_NODE_MAX_IN,
+                self.ConfigProp.QUOTA_NODE_MAX_EG
+        ]:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return None
+            if r == 1:
+                return 64  # practical minimum
+            return math.floor(2**(self.rnd.random() * 35 + 5))  # up to 1 TB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_BPS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 30))  # up to 1 GB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_RATIO:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            if r == 1:
+                return 1
+            return self.rnd.random()
+
+        if prop == self.ConfigProp.THROTTLE_DELAY_MAX_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 25))  # up to ~1 year
+
+        if prop == self.ConfigProp.BAL_WINDOW_MS:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return 1
+            if r == 1:
+                return 2147483647
+            return math.floor(2**(self.rnd.random() * 31))
+
+        if prop == self.ConfigProp.BAL_PERIOD_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 22))  # up to ~1.5 months
+
+        raise Exception(f"Unsupported ConfigProp: {prop}")
+
+    def setUp(self):
+
+        self.config = {}
+        for prop in list(self.ConfigProp):
+            val = self.get_config_parameter_random_value(prop)
+            self.config[prop] = val
+            if not val is None:
+                self.redpanda.add_extra_rp_conf({prop.value: val})
+
+        self.logger.info(
+            f"Initial cluster props: {self.redpanda._extra_rp_conf}")
+        super(ThroughputLimitsSnc, self).setUp()
```

Comment:
in the previous review i left a note about a small experiment that seemed to confirm you could add this into the constructor and still get the behavior you wanted. is this setUp overload doing something special that i'm missing?